File: src/main/kotlin/com/example/tddexamples/OTP/service/RandomNumber.java
/*
package com.example.tddexamples.OTP.service;

import java.util.Random;

public class RandomNumber {

    private char[] CHAR_LIST ={'1','2','3','4','5','6','7','8','9','0'};

    private Integer RANDOM_STRING_LENGTH=6;

    private int length =6;

     public String getRandomString(){

         int randomNum = generateNum();

         StringBuffer randStr = new StringBuffer() ;

         for (int i=0; i<=RANDOM_STRING_LENGTH; i++){

             randStr.append(CHAR_LIST[randomNum]);

         }

         return randStr.toString();

}


    private  int generateNum() {
        String numbers = "1234567890";
        Random random = new Random();
        char[] otp = new char[RANDOM_STRING_LENGTH];

        for(int i = 0; i< RANDOM_STRING_LENGTH ; i++) {
            otp[i] = numbers.charAt(random.nextInt(numbers.length()));
        }

      String otpnew = String.valueOf(otp);

        int num = Integer.parseInt(otpnew);

        return num;
    }



}
*/

File: src/test/kotlin/com/example/tddexamples/RandomNumber.kt
package com.example.tddexamples

import org.junit.Assert
import org.junit.Test
import java.util.*

class RandomNumber {

    private val CHAR_LIST = "1234567890"
    private val RANDOM_STRING_LENGTH = 6


    fun generateRandomString(): String {

        val randStr = StringBuffer()
        for (i in 0 until RANDOM_STRING_LENGTH) {
            val number = getRandomNumber()
            val ch = CHAR_LIST[number]
            randStr.append(ch)
        }
        return randStr.toString()
    }

    /**
     * This method generates random numbers
     * @return int
     */
    private fun getRandomNumber(): Int {
        var randomInt = 0
        val randomGenerator = Random()
        randomInt = randomGenerator.nextInt(CHAR_LIST.length)
        return if (randomInt - 1 == -1) {
            randomInt
        } else {
            randomInt - 1
        }

    }

    @Test
    fun otpTest(){
        var otp = generateRandomString()

        Assert.assertTrue(otp.length==6)
    }

}